tests/scripts/all.sh

```diff
@@ -400,6 +400,18 @@ fi
 
 fi # MemSan
 
+msg "build: cmake build from subdirectory"
+cleanup
+mkdir mbedtls_test_build
```

Comment:
To avoid issues being masked if there's some recursive search for a file in the build process, (now or in the future), I think it would be better to do the build in a directory in the parent directory. e.g.
```
mkdir ../mbedtls_test_build
cd ../mbedtls_test_build
cmake ../<mbedtls>/
```

Although that means you'll have to find the name of the source directory (which could be anything).